kernel/names.mli

```diff
@@ -605,3 +605,9 @@ type evaluable_global_reference =
   | EvalConstRef of Constant.t
 
 val eq_egr : evaluable_global_reference ->  evaluable_global_reference -> bool
+
+(** Located identifiers and objects with syntax. *)
+
+type lident = Id.t CAst.t
```

Comment:
I don't think that these types should be put in the kernel. They refer to AST types that live well upwards.

Reply:
Where do you want to put them? Names is not such a bad place after all I think for them.

Reply:
We are talking like 4 lines; in fact I'd even merge `Names` and `Nameops` and get rid of one module. We have too many modules.

Reply:
Imagine having all identifier-related stuff in a single module, can't be true!

Reply:
Names in already too big for its own good, IMHO... I'd rather split it.

Reply:
I hope we are not leading to a world where 20% of the TCB are `open Foo` statements XD

Reply:
I'd rather have these 4 silly lines not block this PR for very long. I still think that `Names` is not such a bad place for the located objects as IMHO it seems natural that the module defining `Id.t` also defines it's located version.

Noted that this is already used in the next layer `library/` so this is far from "well upwards" but instead it is "immediatedly upwards".

Reply:
OK, OK, fine with me.

Reply:
I understand your concern, but couldn't do better :( but yeah, if we find another place I'll be very happy to move it. Note indeed that Names and Subst have an special place within the kernel as they are used by both the checker and the kernel itself. So in fact they should be placed in a different place, even in clib. We have loads of stuff there that the kernel uses so indeed that's a tricky question.